File: app/src/main/java/com/wildlifeoftianjin/ui/framework/recyclerview/ArticleListAdapter.java
package com.wildlifeoftianjin.ui.framework.recyclerview;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import com.wildlifeoftianjin.model.ArticleOverview;

/**
 * Created by presisco on 2017/5/16.
 */

public class ArticleListAdapter extends DragAppendAdapter<ArticleOverview> {
    public ArticleListAdapter(Context context) {
        super(context);
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return null;
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {

    }

    @Override
    public int getItemCount() {
        return 0;
    }
}

File: app/src/main/java/com/wildlifeoftianjin/ui/framework/recyclerview/ArticleListAdapter.kt
package com.wildlifeoftianjin.ui.framework.recyclerview

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.wildlifeoftianjin.R
import com.wildlifeoftianjin.model.ArticleOverview

/**
 * Created by presisco on 2017/5/16.
 */

class ArticleListAdapter(context: Context, val onClickedArticle: (Int) -> Unit, onFooter: () -> Unit)
    : DragAppendAdapter<ArticleOverview>(context),
        DragAppendAdapter.ActionListener {

    init {
        super.onFooter = onFooter
        onAction = this
    }

    override fun onCreate(inflater: LayoutInflater, parent: ViewGroup): ContentHolder? {
        return ArticleHolder(inflater.inflate(R.layout.listitem_article_overview, parent, false), onClickedArticle)
    }

    override fun onBind(holder: ContentHolder, position: Int) {
    }

    override fun onRecycled(holder: ContentHolder) {
    }

    class ArticleHolder(itemView: View, listener: (Int) -> Unit) : ContentHolder(itemView)
}
